### PR TITLE
[google_maps_flutter_platform_interface] Platform interface changes to support heatmaps

### PR DESCRIPTION
**Pull request by Rexios80**
Prequel to:
- https://github.com/flutter/packages/pull/7313
- https://github.com/flutter/packages/pull/7314
- https://github.com/flutter/packages/pull/7315
- https://github.com/flutter/packages/pull/3257

### CODE REVIEW

**Comment by Rexios80** [comment]
@stuartmorgan 

```console
Finding changed packages relative to "e411afe3fd269001295b19e71856075e624827f5"...
Rewriting references to: google_maps_flutter_platform_interface...
  Modified packages/google_maps_flutter/google_maps_flutter_android/pubspec.yaml
  Modified packages/google_maps_flutter/google_maps_flutter_android/example/pubspec.yaml
  Modified packages/google_maps_flutter/google_maps_flutter/pubspec.yaml
  Modified packages/google_maps_flutter/google_maps_flutter/example/pubspec.yaml
  Modified packages/google_maps_flutter/google_maps_flutter_ios/pubspec.yaml
  Modified packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/pubspec.yaml
  Modified packages/google_maps_flutter/google_maps_flutter_ios/example/shared/maps_example_dart/pubspec.yaml
  Modified packages/google_maps_flutter/google_maps_flutter_ios/example/ios15/pubspec.yaml
  Modified packages/google_maps_flutter/google_maps_flutter_web/pubspec.yaml
  Modified packages/google_maps_flutter/google_maps_flutter_web/example/pubspec.yaml
Running for all packages that have uncommitted changes
```

How can I make the CI not do this?

**Comment by Rexios80** [comment]
Ugh that causes other CI issues. @stuartmorgan please advise on how to resolve this. I see the wiki info about platform interface breaking changes. How do we want to handle this?

**Comment by stuartmorgan** [comment]
Sorry, I missed that in review due to diff collapsing. You don't edit any of the `buildView*` methods, they are legacy for this reason. The platform implementations just add heatmap support in `buildViewWithConfiguration`.

**Comment by Rexios80** [comment]
Should those methods be marked as `@Deprecated` by now?

**Comment by stuartmorgan** [comment]
Yes, you could add those here if you want.

Now that our settings for deprecation analysis have changed we add them as we go; that used to break us, so it required a second pass after everything else was updated, and we often didn't remember to do that in practice (as here).

**Comment by Rexios80** [comment]
@stuartmorgan checks pass! Let's review this again.

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/platform_interface/google_maps_flutter_platform.dart [hunk 1]
```diff
@@ -389,6 +403,7 @@ abstract class GoogleMapsFlutterPlatform extends PlatformInterface {
 
   /// Returns a widget displaying the map view - deprecated, use
   /// [buildViewWithConfiguration] instead.
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/platform_interface/google_maps_flutter_platform.dart:405]: This deprecation-ish comment should be removed; it was what we did when we couldn't use actual `@Deprecation`. So the comment should now just be `/// Returns a widget displaying the map view.`

File: packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/platform_interface/google_maps_flutter_platform.dart [hunk 2]
```diff
@@ -417,6 +432,7 @@ abstract class GoogleMapsFlutterPlatform extends PlatformInterface {
   /// [buildView]. This is for backward compatibility with existing
   /// implementations. Platforms that use the text direction should override
   /// this as the primary implementation, and delegate to it from buildView.
+  @Deprecated('Use buildViewWithConfiguration instead.')
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/platform_interface/google_maps_flutter_platform.dart:435]: Same with lines 425-26 here; this should replace that bit.

**Review by stuartmorgan — APPROVED** [review]
LGTM! Two very small comment nits; no need to request another review for those changes, you can just add the `autosubmit` label once you upload those diffs.

**Pull request merged**